### ISSUE TITLE
No window title when running egui-app on Wayland

### ISSUE THREAD

**Issue by anarsoul** [emilk/egui#992]
**Describe the bug**
There's no window title in window header when egui app is running on Wayland

**To Reproduce**
cd egui
cargo run --bin egui_demo_app

Window header is solid white.

![Screenshot from 2021-12-23 21-48-55](https://user-images.githubusercontent.com/82057/147321779-ea265cc2-a257-4445-95d6-bb5174d95838.png)

**Expected behavior**

Window header contains window title that is returned by epi::App::name

**Screenshots**
See above

**Desktop (please complete the following information):**
 - OS: archlinux
 - Version: rolling release

**Additional context**

I understand that it's a winit/client-toolkit bug, see https://github.com/Smithay/client-toolkit/issues/196

Basically neither winit nor client-toolkit want to mess with font rendering. Yet it affects egui. It would be nice to be able to render window decorations at egui level.

**Comment by parasyte** [comment]
AFAIK, the only cross-platform way to do this is creating the window without decorations and drawing client-side decorations, handling drag and double-click events in platform-specific ways, etc. And doing nothing on Android, iOS, and web platforms because they don't have window decorations anyway (at least the decorations cannot be controlled by `winit`). This is the method used by the Spotify and Discord clients, for example.

On the other hand, doing this client-side decorations thing only for Wayland might be an acceptable way to "plug the leak".

**Comment by emilk** [comment]
A lot of egui users have been making their own decorations, so it is definitely possible (see for instance https://www.youtube.com/watch?v=NtUkr_z7l84). It would be very nice with an official eframe example for how to make your own window decoration.

Related to this: we need to fix https://github.com/emilk/egui/issues/943 and add support for minimizing the native window.

**Comment by emilk** [comment]
There is an example of painting your own window frame in https://github.com/emilk/egui/tree/master/examples/custom_window_frame

**Comment by Krysme** [comment]
> There is an example of painting your own window frame in https://github.com/emilk/egui/tree/master/examples/custom_window_frame

I tried the example on Wayland, and the close button doesn't work. Once I clicked the close button, other buttons stop working. But then after I drag the window for a bit, it seems other buttons ( which control the theme) fall back to normal. However, the close button still doesn't do it's job after dragging.

**Comment by emilk** [comment]
@Krysme have you tried debugging it? Is `close_response.clicked()` ever `true`? Is `frame.quit();` called?

**Comment by flukejones** [comment]
> > There is an example of painting your own window frame in https://github.com/emilk/egui/tree/master/examples/custom_window_frame
> 
> I tried the example on Wayland, and the close button doesn't work. Once I clicked the close button, other buttons stop working. But then after I drag the window for a bit, it seems other buttons ( which control the theme) fall back to normal. However, the close button still doesn't do it's job after dragging.

To do this for my own app I had to resize the title space. What seemed to be happening is the horizontal layout was overlapping the close button. See https://gitlab.com/asus-linux/asusctl/-/blob/main/rog-control-center/src/widgets/top_bar.rs#L50

**Comment by emilk** [comment]
Closed by https://github.com/emilk/egui/pull/1914
